cache-core/src/main/java/io/micronaut/cache/interceptor/CacheInterceptor.java

```diff
@@ -982,6 +996,13 @@ private class CacheOperation {
             }
         }
 
+        private <T extends Annotation> boolean hasConditional(@NonNull List<AnnotationValue<T>> annotationValues) {
+            if (CollectionUtils.isEmpty(annotationValues)) {
+                return false;
+            }
+            return annotationValues.stream().anyMatch(av -> av.isPresent(MEMBER_CONDITION));
```

Comment:
does the default value of the condition member "" evaluate as true always due to` av.isPresent(MEMBER_CONDITION)` ? Should we evaluate `condition`value is not an empty string with `StringUtils.isNotEmpty`?

Reply:
or does AnnotationValue::isPresent evaluates to false for empty strings?

Reply:
this should use `av.isPresent(MEMBER_CONDITION)`